src/rdkafka_cgrp.c

```diff
@@ -5321,6 +5372,7 @@ rd_kafka_cgrp_subscription_set(rd_kafka_cgrp_t *rkcg,
                         rkcg->rkcg_consumer_flags |=
                             RD_KAFKA_CGRP_CONSUMER_F_SUBSCRIBED_ONCE |
                             RD_KAFKA_CGRP_CONSUMER_F_SEND_NEW_SUBSCRIPTION;
+                        rd_kafka_cgrp_maybe_clear_heartbeat_failed_err(rkcg);
```

Comment:
This is more specific to `TopicAuthorizationError`, We should not clear it here. If we add a new error there, this is not the correct place to clear the error. Moreover, if there is a new subscription set, we send the new HB and then if there is no issue, the error will be cleared there.

Reply:
This just ensures that when you change the subscription you receive a new error on first HB after the subscription if it's still on error and don't have to wait for at least 5 seconds. That applies to other kind of future errors as well.